### DISCUSSION TITLE
Requesting change in the Documentation for NGINX Reverse Proxy

### DISCUSSION THREAD

**Discussion by paulkoehlerdev** [authelia/authelia#6170] [General]
Hi!

I have struggled a bit with the oCIS Desktop Client, an oCIS configured to use Authelia as a OIDC provider.
There is a pitfall baked into the example configuration for Authelia and NGINX with these kinds of setups.

These kinds of setups include all OIDC configurations, that depend on a `http`-scheme `redirect_url` for anything.

The offending line seems to be:
```nginx
proxy_redirect http:// $scheme://;
```
[(link)](https://github.com/authelia/authelia/blob/9d4b303f90182b35088488e78fa19581df39b95c/docs/content/en/integration/proxies/nginx.md?plain=1#L376)

I would suggest at least adding a warning to this line:
```nginx
# Warning: If you are using an OIDC client with an http redirect_uri, this will break the flow.
proxy_redirect http:// $scheme://;
```
[(link to change in fork)](https://github.com/paulkoehlerdev/authelia/commit/d99d187f2b520f51db09b238c937953dbcab91e7)

**Comment by james-d-elliott** [comment]
I'd generally recommend not securing a OIDC protected app with the forward authentication flow as well. Is there a situation where you don't feel this would work?

**Reply by paulkoehlerdev** [reply]
There are reasons to do this, as there can be problems in the OIDC implementation of some software, which could lead to insecure pages. This would allow to share users and not trust many services.

Other then that. The snippet is also used in the `auth.conf` [(reference)](https://github.com/authelia/authelia/blob/9d4b303f90182b35088488e78fa19581df39b95c/docs/content/en/integration/proxies/nginx.md?plain=1#L233C1-L236C6), where this leads to problems, even if you don't use the `auth_request` functionality of NGINX for the OIDC service.

**Reply by james-d-elliott** [reply]
Ahh, I think I see... because that line prevents redirections to http:// URLs I believe, which I think prevents the redirect to the http://localhost address. I believe we can just outright remove that line since we bake in protection against this from within Authelia.
